pocs/focuser/focuser.py

```diff
@@ -387,13 +392,17 @@ def _autofocus(self,
                                              focus_positions[i], i, self._camera.file_extension)
             thumbnail = self._camera.get_thumbnail(
                 seconds, file_path, thumbnail_size, keep_file=keep_files)
-            thumbnail = focus_utils.mask_saturated(thumbnail)
+            masks[i] = focus_utils.mask_saturated(thumbnail).mask
             if dark_thumb is not None:
                 thumbnail = thumbnail - dark_thumb
-            # Calculate focus metric
-            metric[i] = focus_utils.focus_metric(
-                thumbnail, merit_function, **merit_function_kwargs)
-            self.logger.debug("Focus metric at position {}: {}".format(position, metric[i]))
+            thumbnails[i] = thumbnail
+
+        master_mask = masks.any(axis=0)
+        master_mask = binary_dilation(master_mask, iterations=dilations)
```

Comment:
Any chance this will ever mask out too much?

Reply:
It might mask slightly more than absolutely necessary under some circumstances. The size of the masked regions around saturated pixels is controlled by the number of dilations, which I've made a (secret, undocumented, inaccessible) adjustable parameter.

For any sane value of `dilations` it's never going to mask so much that it interferes with the focusing though.  The only way it would mask a significant proportion of the image is if there was a *lot* of saturated stars (i.e. exposure time is too long), which would cause the focusing to fail regardless.

I should probably made the number of dilations (and the smoothing parameter for the spline fit) into documentated, accessible parameters that can be adjusted via the config file, though, like the rest of the autofocus algorithm parameters.

Reply:
Great, thanks for explanation. If you can't make the dilations accessible immediately but want to test this tonight that is okay.